crates/fluvio/src/producer/error.rs

```diff
@@ -28,3 +28,18 @@ pub enum ProducerError {
     #[error("the batch enqueue timeout limit reached")]
     BatchQueueWaitTimeout,
 }
+
+impl ProducerError {
+    /// Suggest possible fixes to the errors
+    pub fn user_message(&self) -> Option<String> {
+        match self {
+            Self::RecordTooLarge(size) => {
```

Comment:
You can add direct into Error itself using `thiserror`

Reply:
I though that that is for the display representation, therefore not sure if it was a good place to put the suggestion.Updating this to use it

Reply:
A single unified message would be good enough